### PR TITLE
auth: fix rounding inaccuracy in latency statistics

### PR DESCRIPTION
**Pull request by mind04**
### Short description
The latency statistics are off by a lot. This pull is changing the type of the avg_latency variable from int to double to avoid rounding issues in te calculation.

### Checklist
I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by Habbie** [comment]
This is the kind of thing that raises questions after an upgrade - can you add an upgrade note?

File: pdns/common_startup.cc [hunk 1]
```diff
@@ -399,7 +399,7 @@ static void sendout(std::unique_ptr<DNSPacket>& a)
   N->send(*a);
 
   int diff=a->d_dt.udiff();
-  avg_latency=(int)(0.999*avg_latency+0.001*diff);
+  avg_latency=(0.999*avg_latency+0.001*diff);
```
Comment (omoerbeek) [inline pdns/common_startup.cc:402]:
```suggestion
  avg_latency=0.999*avg_latency+0.001*diff;
```

File: pdns/common_startup.cc [hunk 2]
```diff
@@ -484,7 +484,7 @@ try
         cached.commitD(); // commit d to the packet                        inlined
         NS->send(cached); // answer it then                              inlined
         diff=question.d_dt.udiff();
-        avg_latency=(int)(0.999*avg_latency+0.001*diff); // 'EWMA'
+        avg_latency=(0.999*avg_latency+0.001*diff); // 'EWMA'
```
Comment (omoerbeek) [inline pdns/common_startup.cc:487]:
```suggestion
        avg_latency=0.999*avg_latency+0.001*diff; // 'EWMA'
```

**Comment by Habbie** [comment]
The 4.4 backport is merged - this PR needs documentation before we release 4.4.

**Comment by Habbie** [comment]
> This is the kind of thing that raises questions after an upgrade - can you add an upgrade note?

I pushed one. Please check :)

**Pull request merged**